### PR TITLE
chore: run clippy against all targets

### PR DESCRIPTION
**Pull request by 0xPoe**
This would help us to check if there is any clippy warning in the tests.

I found it after https://github.com/tokio-rs/console/pull/544. There are some unused imports in our test code.

Use `cargo clippy --workspace --all-targets --no-deps -- -D warnings` to test all targes which include the test targets.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (hds) [inline .github/workflows/ci.yml]: Could we avoid the unrelated formatting changes? I think this might be your editor, is that right?

**Review by hds — CHANGES_REQUESTED** [review]
Good idea! Would be good to avoid the unrelated formatting changes in the GitHub Actions file though.

File: console-subscriber/tests/spawn.rs
Comment (hawkw) [inline console-subscriber/tests/spawn.rs:15]: in general, it's nice if allow attributes for lints can have a comment explaining why they're allowed in this instance. it would be nice if you're able to add one here, although it's not a huge deal.

File: console-subscriber/tests/support/task.rs
Comment (hawkw) [inline console-subscriber/tests/support/task.rs]: nit, take out or leave it: maybe we should allow this globally for the whole `impl` block?

**Review by hawkw — COMMENTED** [review]
overall, this looks good, thanks for the lint fixes! i agree with @hds that we should undo the YAML formatting changes before merging.

File: console-subscriber/tests/spawn.rs
Comment (0xPoe) [reply console-subscriber/tests/spawn.rs:15]: Good suggestion! Added.

**Review by 0xPoe — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: console-subscriber/tests/support/task.rs
Comment (0xPoe) [reply console-subscriber/tests/support/task.rs]: Make sense. Moved.

File: .github/workflows/ci.yml
Comment (0xPoe) [reply .github/workflows/ci.yml]: It was reformated by my IDE. Those changes were dropped.

**Review by hawkw — APPROVED** [review]
looks good to me, thank you!

**Comment by 0xPoe** [comment]
@hds This PR needs your approval before I can merge it. Could you please take a look? Thanks!
image

**Review by hds — APPROVED** [review]
Thanks for adding this, great idea!

**Comment by hds** [comment]
> @hds This PR needs your approval before I can merge it. Could you please take a look? Thanks!
> 
> image
> 
> 

Sorry! I didn't realize I was blocking this one. 🤦‍♀️

**Comment by 0xPoe** [comment]
> Sorry! I didn't realize I was blocking this one. 🤦‍♀️

No worry! Thank you ❤️

**Pull request merged**
